tests/test_api.py

```diff
@@ -3659,3 +3660,357 @@ def test_operations(self):
+                   from_date=datetime.datetime(1900, 1, 1),
+                   to_date=datetime.datetime(2017, 6, 1))
+
+        api.add_identity(self.ctx, 'scm', email='jsmith@bitergia')
+        api.add_identity(self.ctx,
+                         'phabricator',
+                         email='jsmith-phab@bitergia',
+                         uuid='e8284285566fdc1f41c8a22bb84a295fc3c4cbb3')
+        api.update_profile(self.ctx,
+                           uuid='e8284285566fdc1f41c8a22bb84a295fc3c4cbb3', name='John Smith',
+                           email='jsmith@profile-email', is_bot=True, country_code='US')
+        api.enroll(self.ctx,
+                   'e8284285566fdc1f41c8a22bb84a295fc3c4cbb3', 'Bitergia',
+                   from_date=datetime.datetime(2017, 6, 2),
+                   to_date=datetime.datetime(2100, 1, 1))
+
+        api.add_identity(self.ctx,
+                         'scm',
+                         email='jsmith@libresoft',
+                         uuid='e8284285566fdc1f41c8a22bb84a295fc3c4cbb3')
+        api.add_identity(self.ctx,
+                         'phabricator',
+                         email='jsmith2@libresoft',
+                         uuid='e8284285566fdc1f41c8a22bb84a295fc3c4cbb3')
+        api.add_identity(self.ctx,
+                         'phabricator',
+                         email='jsmith3@libresoft',
+                         uuid='e8284285566fdc1f41c8a22bb84a295fc3c4cbb3')
+        api.update_profile(self.ctx,
+                           uuid='e8284285566fdc1f41c8a22bb84a295fc3c4cbb3', name='John Smith',
+                           email='jsmith@profile-email', is_bot=False, country_code='US')
+
+    def test_unmerge_identities(self):
+        """Check whether it unmerges one identity from its parent unique identity"""
+
+        uidentities = api.unmerge_identities(self.ctx,
+                                             uuids=['67fc4f8a56aa12ab981d2a4c1de065bb9936c9f6'])
+
+        # Tests
+        self.assertEqual(len(uidentities), 1)
+
+        uidentity = uidentities[0]
+        self.assertIsInstance(uidentity, UniqueIdentity)
+        self.assertEqual(uidentity.uuid, '67fc4f8a56aa12ab981d2a4c1de065bb9936c9f6')
+
+        profile = uidentity.profile
+
+        self.assertEqual(profile.email, 'jsmith-git@example')
+        self.assertEqual(profile.is_bot, False)
+        self.assertIsNone(profile.name)
+        self.assertIsNone(profile.gender)
+        self.assertIsNone(profile.gender_acc)
+
+        self.assertIsNone(profile.country_id)
+        self.assertIsNone(profile.country)
+
+        identities = uidentity.identities.all()
+        self.assertEqual(len(identities), 1)
+
+        id1 = identities[0]
+        self.assertEqual(id1.id, '67fc4f8a56aa12ab981d2a4c1de065bb9936c9f6')
+        self.assertEqual(id1.email, 'jsmith-git@example')
+        self.assertEqual(id1.source, 'git')
+
+        enrollments = uidentity.enrollments.all()
+        self.assertEqual(len(enrollments), 0)
```

Comment:
I think you should check here that the other identities weren't move from the original unique identity and that other things as the profile or the enrollments didn't change either.

Reply:
Done, I have checked the identities, profile and enrollment from the former parent unique identity.